server/src/main/java/org/elasticsearch/health/node/DiskHealthInfo.java

```diff
@@ -8,20 +8,66 @@
 
 package org.elasticsearch.health.node;
 
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.health.HealthStatus;
 
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
 /**
  * The health status of the disk space of this node along with the cause.
  */
-record DiskHealthInfo(HealthStatus healthStatus, Cause cause) {
+public record DiskHealthInfo(HealthStatus healthStatus, Cause cause) implements Writeable {
     DiskHealthInfo(HealthStatus healthStatus) {
         this(healthStatus, null);
     }
 
-    enum Cause {
-        NODE_OVER_HIGH_THRESHOLD,
-        NODE_OVER_THE_FLOOD_STAGE_THRESHOLD,
-        FROZEN_NODE_OVER_FLOOD_STAGE_THRESHOLD,
-        NODE_HAS_NO_DISK_STATS
+    public DiskHealthInfo(StreamInput in) throws IOException {
+        this(HealthStatus.from(in), in.readOptionalWriteable(Cause::from));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        healthStatus.writeTo(out);
+        out.writeOptionalWriteable(cause);
+    }
+
+    public enum Cause implements Writeable {
+        NODE_OVER_HIGH_THRESHOLD((byte) 0),
+        NODE_OVER_THE_FLOOD_STAGE_THRESHOLD((byte) 1),
+        FROZEN_NODE_OVER_FLOOD_STAGE_THRESHOLD((byte) 2),
+        NODE_HAS_NO_DISK_STATS((byte) 3);
+
+        private static final Map<Byte, Cause> values = Arrays.stream(Cause.values())
```

Comment:
the enum  has the `values()` already. there's no need for this I don't think

Reply:
I did this only for retrieval, which is not necessary if we are using the `readEnum`. I removed it.